src/annmas/segment/command.py

```diff
@@ -47,60 +84,334 @@ def main(model, threads, output_bam, input_bam):
         m.from_yaml(model)
         logger.info(f"Using pretrained annotation model {model}")
     else:
-        logger.info(f"Using default annotation model")
+        logger.info("Using default annotation model")
+
+    if do_simple_splitting:
+        logger.info("Using simple splitting mode.")
+    else:
+        logger.info("Using bounded region splitting mode.")
 
     num_reads_annotated = 0
     num_sections = 0
-
-    pysam.set_verbosity(0)  # silence message about the .bai file not being found
-    with pysam.AlignmentFile(input_bam, "rb", check_sq=False, require_index=False) as bam_file:
-
-        with concurrent.futures.ThreadPoolExecutor(max_workers=threads) as executor,\
-                tqdm.tqdm(desc="Progress", unit=" read", colour="green", file=sys.stdout) as pbar:
-
-            future_to_segmented_read = {executor.submit(segment_read, r, m): r for r in bam_file}
-
-            # Get our header from the input bam file:
-            out_bam_header_dict = bam_file.header.to_dict()
-
-            # Add our program group to it:
-            pg_dict = {'ID': 'annmas-segment-0.0.1', 'PN': 'annmas', 'VN': '0.0.1',
-                       'DS': 'Apply annotation and segmentation model to BAM file.', 'CL': " ".join(sys.argv)}
-            if "PG" in out_bam_header_dict:
-                out_bam_header_dict["PG"].append(pg_dict)
+    num_segments = 0
+
+    # Placeholder for split out file:
+    # NOTE: There's probably a more pythonic way to do this...
+    split_bam_out = None
+
+    try:
+        pysam.set_verbosity(0)  # silence message about the .bai file not being found
+        with pysam.AlignmentFile(
+            input_bam, "rb", check_sq=False, require_index=False
+        ) as bam_file:
+
+            with concurrent.futures.ThreadPoolExecutor(
+                max_workers=threads
+            ) as executor, tqdm.tqdm(
+                desc="Progress", unit=" read", colour="green", file=sys.stdout
+            ) as pbar:
+
+                future_to_segmented_read = {
+                    executor.submit(_segment_read, r, m): r for r in bam_file
```

Comment:
Yeah - I thought about doing it this way first, but as I was looking through examples I saw the one I implemented and it seemed more pythonic.  I didn't realize it would have to parse the whole file before starting to dispatch the data, though.  I'll move to the queue-based thread workers.